### PR TITLE
Updated uWSGI to properly use threads and handle static assets

### PR DESCRIPTION
**Pull request by blarghmatey**
**Added static asset handling to uWSGI**
Updated uWSGI config to handle loading static assets to decrease load on Django

**Exposed environment vars for uWSGI process/thread counts**
Added environment variables for being able to set the number of
processes and threads used in uWSGI.

**Added pgbouncer buildpack**
Added pgbouncer buildpack to account for increased DB connections from
multiple threads.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -20,11 +20,12 @@ web:
     /bin/bash -c '
     sleep 3 &&
     python3 manage.py migrate &&
-    ./with_host.sh python3 manage.py runserver 0.0.0.0:8079'
+    uwsgi uwsgi.ini'
   volumes:
     - .:/src
   environment:
     DEBUG: 'True'
+    DEV_ENV: True
```
Comment (giocalitri) [inline docker-compose.yml]: please quote this variable

**Comment by giocalitri** [comment]
please format the logging so that the requested endpoint are more visible
```
web_1     | {address space usage: 449560576 bytes/428MB} {rss usage: 63332352 bytes/60MB} [pid: 14|app: 0|req: 2/5] 192.168.99.1 () {40 vars in 792 bytes} [Thu Nov  3 18:13:40 2016] GET /api/v0/profiles/gio/ => generated 1351 bytes in 60 msecs (HTTP/1.1 200) 4 headers in 135 bytes (1 switches on core 0)
```

File: uwsgi.ini [hunk 2]
```diff
@@ -10,8 +21,15 @@ pidfile=/tmp/micromasters-mast.pid
...
+offload-threads = 2
+thunder-lock =
 if-env = HTTP_AUTH_CREDENTIALS
 plugins = python,router_basicauth
 route = ^/api/v\d+/order_fulfillment/? continue:
 route = ^/.* basicauth:MicroMasters,%(_)
 endif =
+if-env = DEV_ENV
+python-autoreload = 1
+endif =
```
Comment (giocalitri) [inline uwsgi.ini]: please add a whitespace add end of the file

**Comment by Ferdi** [comment]
Why not use s3 for static assets ?

**Comment by blarghmatey** [comment]
@Ferdi there's no particular reason not to use S3. We can tackle that in a separate PR. For now I just wanted to help alleviate some of the work that needed to be done by the Python code.

**Comment by giocalitri** [comment]
👍 
please remember to squash

**Pull request merged**